### PR TITLE
GH-82: Extend DupFinder properties in ToolSettings

### PR DESCRIPTION
**Pull request by kcamp**
Adds in additional properties to pass along to `DupFinderSettings`.

It may also be worth considering creating nested classes to start to delineate the various settings?  Just adding these new parameters + properties starts to feel messy.  If that's something that is desired, I could potentially re-work this PR to include it.

### CODE REVIEW

File: Cake.Recipe/Content/toolsettings.cake [hunk 1]
```diff
@@ -1,6 +1,9 @@
 public static class ToolSettings
 {
     public static string[] DupFinderExcludePattern { get; private set; }
+    public static string[] DupFinderExcludeFilesByStartingCommentSubstring { get; private set; }
+    public static int? DupFinderDiscardCost { get; private set; }
+    public static bool? DupFinderThrowExceptionOnFindingDuplicates { get; set; }
```
Comment (gep13) [inline Cake.Recipe/Content/toolsettings.cake]: Any reason that this isn't a `private set` in the same way as the other properties?

**Comment by gep13** [comment]
Relates to #82

File: Cake.Recipe/Content/toolsettings.cake (hunk 1 above)
Comment (kcamp) [inline Cake.Recipe/Content/toolsettings.cake]: No reason, just an oversight.  Will fix it up and refresh.

**Comment by kcamp** [comment]
@gep13 Fixed up the `private` setter and refreshed the PR.  Are you interested in attempting to introduce additional separation between the various ToolSettings types via nesting classes or similar?

**Comment by gep13** [comment]
> @kcamp said...
Are you interested in attempting to introduce additional separation between the various ToolSettings types via nesting classes or similar?

Off the top of my head, I am not against it.  Can I suggest you raise an issue with what you were thinking, and we can discuss it over there.

Thanks again for making this change.

**Comment by kcamp** [comment]
No problem.  I will put together something along the lines of what I am thinking could be useful and we can move it to there.  Thanks!  👍

**Pull request merged**